x/dualstaking/README.md

```diff
@@ -0,0 +1,138 @@
+Providers in lava give services to consumers, they compete for giving services with other providers by staking tokens.
+Since a lot of tokens are expected to be staked by providers lava give providers the power to add security to the chain with dualstaking.
+
+This document focuses on the dualstaking technical aspects. For more information on those, please visit Lava's official website.
+
+## Contents
+* [Concepts](#concepts)
+    * [Delegation](#delegation)
+    * [EmptyProvider](#emptyprovider)
+    * [Dualstaking](#dualstaking)
+        * [Dualstaking](#validatordelegation)
+        * [Dualstaking](#validatorunbonding)
+        * [Dualstaking](#validatorslashing)
+        * [Dualstaking](#providerdelegation)
+        * [Dualstaking](#providerunbonding)
+    * [Hooks](#hooks)
+    * [RedelegateFlag](#redelegateflag)
+    * [Rewards](#rewards)
+* [Parameters](#parameters)
+* [Queries](#queries)
+* [Transactions](#transactions)
+* [Proposals](#proposals)
+* [Events](#events)
+
+## Concepts
+
+### Delegation
+
+Dualstaking intoduces to lava provider delegations. provder delegations lets users to delegate their tokens to a specific provider (similar to validators), contribute to their success and claim some of the rewards awarded to the provider.
+When a provider is staking tokens he is creating a self-delegation entry.
+Whenever a provider recieves rewards, all the delegators will be aligible for a portion of it proportionate to their delegation amount and the commision rate set by the provider.
+
+### EmptyProvider
+
+The empty provider is a place holder for provider delegations that are issued by the staking module. the user can choose to redelegate from this provider to an actual provider.
+
+### Dualstaking
+
+Dualstaking exists to give power to providers the same as validators. whenever a provider stakes tokens the same amount is also staked to a validator.
+Dualstaking does this by implementing providers delegators (similar to validators delegators) and using hooks to from the staking module.
+For every validator delegation exists a parralel provider delegation and vice versa.
+When a provider stakes his tokens, he is a self delegating provider and delegating to a validator with the same amount. This give him gov influence and support a validator of his choosing which in turn boosts the security of the chain (this is also true to providers delegators).
+The same way happens with validators, when a validator stakes his tokens (or a delegator), using hooks a delegation is created to an empty provider. Than using the dualstaking module he can redelegate from the empty provider to an actual provider.
+Since the module uses hooks to achieve this functionality it does not need to move coins, the staking module does it for him.
+For a specific address, at all times the amount of validators delegation is equal to providers delegations.
+The following are use cases of the dualstaking module:
+
+#### Validator delegation
```

Comment:
for all cases: fix capital letters on sentence start, bad spelling and "." on sentence end

Reply:
done